includes/class-sensei-course.php

```diff
@@ -4462,6 +4462,10 @@ public function has_sensei_blocks( $course = null ) {
 	 * @return String
 	 */
 	public static function alter_redirect_url_after_enrolment( $url, $post ) {
+		// Only redirect to the lesson if the course is published.
+		if ( 'publish' !== $post->post_status ) {
```

Comment:
When taking a course as admin and the course is not published, it should not change the redirect. The user needs to continue on the course page to see a notice. See:
https://github.com/Automattic/sensei/blob/d53e6a4f1533cf5166fae99bafd4b012dd5f91cf/includes/class-sensei-frontend.php#L1261
https://github.com/Automattic/sensei/blob/d53e6a4f1533cf5166fae99bafd4b012dd5f91cf/includes/blocks/class-sensei-course-outline-course-block.php#L69-L71